### PR TITLE
Update Skyblock Member and Skyblock Profile

### PR DESCRIPTION
**Pull request by Johannes7k75**
**Please describe changes**

- The last active skyblock profile has now an selected attribute.
- Skyblock members have a few more values
  - experience
  - hotm
  - trophyFish
  - highestMagicalPower
  - Slayers: enderman, blaze and vampire
  - bestiary Level

- [x] I've added new features. (methods or parameters)
- [x] I've added jsdoc and typings.
- [ ] I've fixed bug. (*optional* you can mention a issue if there is one)
- [ ] I've corrected the spelling in README, documentation, etc.
- [x] I've tested my code. (`npm run test`)

### CODE REVIEW

**Comment by Johannes7k75** [comment]
@Kathund if created a new pull request, because i have added more things.

**Comment by Kathund** [comment]
> @Kathund if created a new pull request, because i have added more things.

Give me like a week i will check this pr

File: .prettierrc
Comment (Kathund) [inline .prettierrc:1]: Please add a print width of 210 to keep with the eslint rules

**Review by Kathund — CHANGES_REQUESTED** [review]

File: .prettierrc
Comment (Johannes7k75) [reply .prettierrc:1]: Is this file even necessary?
This file doesn't even exists on the main repo.

Comment (Kathund) [reply .prettierrc:1]: depends if u want to keep the file.
Prettier is very useful in large projects like this. Im bringing it in #429 so you can keep it if you want but the choice will come down to you

Comment (Johannes7k75) [reply .prettierrc:1]: You mean #428 or? Print Width is fixed

Comment (Kathund) [reply .prettierrc:1]: I would never mess up the pr link shhhhh

File: src/structures/SkyBlock/SkyblockMember.js [hunk 1]
```diff
@@ -69,6 +71,26 @@ class SkyblockMember {
...
+    /**
+     * Experience
+     * @type {number}
+     */
+    this.experience = data.m.leveling?.experience ?? 0;
+    /**
+     * Heart of the Mountain - MiningSkill
+     * @returns {number}
+     */
+    this.hotm = data.m.mining_core?.experience ?? 0;
```
Comment (Kathund) [inline src/structures/SkyBlock/SkyblockMember.js]: Can you please explain what the number means? im testing on the user lighthth and it returns 300k+ and im unsure what it means? its not there hotm total xp coz i check so im unsure

File: src/structures/SkyBlock/SkyblockMember.js
Comment (Kathund) [inline src/structures/SkyBlock/SkyblockMember.js:1]: At line [209](https://github.com/Kathund/hypixel-api-reborn/blob/8f85d8a743c9130699a1ba16ffdc23d1457a7c29/src/structures/SkyBlock/SkyblockMember.js#L209) Is it possible to make it return the stats for kills/deaths like this?

```
{
  kills: {
    total: number,
    mobName: number,
    mobName: number,
    mobName: number,
    mobName: number,
    },
  deaths: {
    total: number,
    mobName: number,
    mobName: number,
    mobName: number,
    mobName: number,
    },
   allOtherStatsEct: :shrug:
 }
 ```

**Review by Kathund — CHANGES_REQUESTED** [review]

File: src/structures/SkyBlock/SkyblockMember.js (hunk 1 above)
Comment (Johannes7k75) [reply src/structures/SkyBlock/SkyblockMember.js]: Its the user has an experience for hotm. [See](https://wiki.hypixel.net/Heart_Of_The_Mountain#Leveling) Cumulative, for the hotm tier

File: src/structures/SkyBlock/SkyblockMember.js
Comment (Johannes7k75) [reply src/structures/SkyBlock/SkyblockMember.js:1]: Sure

**Review by Kathund — APPROVED** [review]

File: src/structures/SkyBlock/SkyblockMember.js (hunk 1 above)
Comment (Kathund) [reply src/structures/SkyBlock/SkyblockMember.js]: If possible can you please make it return in an object like how the skills return?

**Review by Kathund — CHANGES_REQUESTED** [review]

File: src/structures/SkyBlock/SkyblockMember.js (hunk 1 above)
Comment (Johannes7k75) [reply src/structures/SkyBlock/SkyblockMember.js]: Yes it returns now a skill Like object

File: src/structures/SkyBlock/SkyblockMember.js
Comment (Johannes7k75) [reply src/structures/SkyBlock/SkyblockMember.js:1]: made it now so that in kills and deaths are only mob names

**Review by Kathund — APPROVED** [review]

**Comment by Kathund** [comment]
@StavZ Tested on 18de187c988837aac1a80e443528f1f4f3bcc9a2 Everything returns correctly and looks good!

**Comment by Johannes7k75** [comment]
@Kathund on the commit is one issue getMemberStats() has as param obg but it should be obj.
So stats should be undefined or so.

**Comment by Kathund** [comment]
> @Kathund on the commit is one issue getMemberStats() has as param obg but it should be obj.
> So stats should be undefined or so.

should be fine since I see you fixed it and I only tested on that commit

**Comment by Johannes7k75** [comment]
any followup?

**Comment by Kathund** [comment]
> any followup?

There's nothing more that I can do its down to someone who has write access to this repo

**Pull request merged**